packages/password-input/src/PasswordInput/PasswordInput.tsx

```diff
@@ -123,30 +135,32 @@ export const PasswordInput = React.forwardRef<
                     inputDisabledBaseStyles,
                     inputDisabledThemeStyles[theme],
                   )]: disabled,
-                  [inputIconSizeStyles[size]]:
-                    !hasNotifications && state !== State.None,
                 },
               )}
               onChange={handleChange}
               readOnly={disabled}
               {...rest}
             />
-            {/* If a custom message container is used, an icon will render inside the input to represent the state of the input. In the case that stateNotification === `none`, no icon will appear. */}
-            {!hasNotifications && state !== State.None && (
-              <InputIcon state={state} size={size} disabled={disabled} />
-            )}
             <PasswordToggle
               showPassword={showPassword}
               handlePasswordToggleClick={handlePasswordToggleClick}
               size={size}
             />
           </div>
-          {hasNotifications && (
+          {showStateNotifications && (
             <StateNotifications
               id={descriptionId}
               notifications={stateNotifications as Array<NotificationProps>}
             />
           )}
+          {showStateFeedback && (
+            <StateFeedback
+              id={descriptionId}
+              state={state}
+              errorMessage={errorMessage}
```

Comment:
What happens if the consumer passes a blank string?

Reply:
consumer can pass an empty string if they don't want an error message to render. In that case, only the warning icon will render